jsonclient/client.go

```diff
@@ -41,6 +44,69 @@ type JSONClient struct {
 	httpClient *http.Client          // used to interact with the server via HTTP
 	Verifier   *ct.SignatureVerifier // nil for no verification (e.g. no public key available)
 	logger     Logger                // interface to use for logging warnings and errors
+
+	multiplier int64
+	until      *time.Time
+	mu         sync.RWMutex
+}
+
+const (
+	maxBackoff = time.Second * 128
+	jitter     = 250
+)
+
+func (c *JSONClient) setBackoff(override *time.Duration) time.Duration {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.until != nil {
+		return time.Until(*c.until)
```

Comment:
Shouldn't this take `override` into account?  If we're already retrying and so have a `c.notBefore` of (say) 2s away, we should still respect a subsequent `Retry-After`